### DISCUSSION TITLE
REST API wrong Image

### DISCUSSION THREAD

**Discussion by A380Coding** [jomjol/AI-on-the-edge-device#2037] [General] [answered]
When using rest API to check status with "statusflow", the status is "Flow finished" but sometimes I get the image that shows a meter saying "taking image". Under which conditions can that happen? Using alg_roi.jpg Image.

**Comment by Slider0007** [comment]
alg_roi.jpg will be overwitten with "taking image" whenever the next round starts. Maybe you call the alg_roi.jpg too late (or it takes too long to deliver) and new round is already started.

Note: After the state "Aligning" is finished the alg_roi.jpg is already complete processed.

**Comment by A380Coding** [comment, marked as answer]
Selecting "Flow finished" to ensure that calculation is over and not add load for transferring the data. Also to get values and image in sync.

**Reply by Slider0007** [reply]
I agree that's the best approach. What interval are you using? How long is the state "flow finshed"?

**Reply by A380Coding** [reply]
Interval is 5 mins. From image to finished ~50s. So about 4 mins in "flow finished". I guess my request is currently at the edge of next round as you said before. But I do not yet see anything to check to avoid reading value and image at that time.

**Reply by Slider0007** [reply]
I'd suggest to fetch the image and values right when "flow finshed" gets triggered, so you have 4 minutes time.

**Reply by A380Coding** [reply]
How do I get to know when that is? Polling to get the transition to "flow finished"? :(

**Reply by Slider0007** [reply]
Either you use MQTT, there states and values get pushed.
Using REST you need to poll the status. In your case every 2-3min would be sufficient.
